Source/Provers/SMTLib/SMTLibBatchTheoremProver.cs

```diff
@@ -181,10 +185,12 @@ private async Task<SolverOutcome> CheckSat(CancellationToken cancellationToken)
           }
         }
 
-        var modelSExp = responseStack.Pop();
-        errorModel = ParseErrorModel(modelSExp);
+        if (responseStack.Count > 0) {
+          var modelSExp = responseStack.Pop();
+          errorModel = ParseErrorModel(modelSExp);
+        }
 
-        if (options.LibOptions.ProduceUnsatCores) {
+        if (options.LibOptions.ProduceUnsatCores && responseStack.Count > 0) {
```

Comment:
Shouldn't we communicate some kind of error explicitly if `ProduceUnsatCores` is on but `responseStack` is empty?